src/app/features/psbt-signer/hooks/use-psbt-signer.tsx

```diff
@@ -40,6 +40,7 @@ export function usePsbtSigner() {
           // be the account publicKey
           if (taprootSigner && witnessOutputScript?.type === 'tr' && !input.tapInternalKey) {
             input.tapInternalKey = taprootSigner.payment.tapInternalKey;
+            tx.updateInput(idx, input);
```

Comment:
We definitely need to mutate and change? Could we equally spread operate and add the key instead of mutating first?

Reply:
I guess I don't understand not doing this when this is how his examples are shown in the lib? If we are an `updater` to the PSBT, which we are in this case, I think this is how we want to do it?

https://github.com/paulmillr/scure-btc-signer#inputs

Reply:
```suggestion
            tx.updateInput(idx, { ...input, tapInternalKey: taprootSigner.payment.tapInternalKey });
```

this is what I meant. I don't see why you mutate the existing one first, versus just passing in new one?

Reply:
O, gotcha, ok. Will change then merge.